File: spec/java/src/io/kaitai/struct/specwrite/TestRepeatEosBytes.java
package io.kaitai.struct.specwrite;

import io.kaitai.struct.ByteBufferKaitaiStream;
import io.kaitai.struct.ConsistencyError;
import io.kaitai.struct.KaitaiStruct.ReadWrite;
import io.kaitai.struct.testwrite.RepeatEosBytes;

import java.util.ArrayList;
import java.util.Arrays;

import org.testng.annotations.Test;

public class TestRepeatEosBytes extends CommonSpec {
    @Test(expectedExceptions = ConsistencyError.class, expectedExceptionsMessageRegExp = "Check failed: records, expected: 0, actual: 3")
    public void testCheckLongerIo() throws Exception {
        RepeatEosBytes r = new RepeatEosBytes();

        r.setRecords(new ArrayList<>(
            Arrays.asList(
                new byte[] { -24, -70, -86, -86, -86 },
                new byte[] { -6, -98, -72, -86, -86 },
                new byte[] { -86, 85, 85, 85, 85 }
            )
        ));

        r._check();
        r._write(new ByteBufferKaitaiStream(15 + 3));
    }

    @Override
    protected Class<? extends ReadWrite> getStructClass() {
        return RepeatEosBytes.class;
    }

    @Override
    protected String getSrcFilename() {
        return "repeat_until_process.bin";
    }
}

File: spec/python/specwrite/test_repeat_eos_bytes.py
import io
import unittest
from kaitaistruct import ConsistencyError, KaitaiStream
from specwrite.common_spec import CommonSpec

from testwrite.repeat_eos_bytes import RepeatEosBytes

class TestRepeatEosBytes(CommonSpec.Base):
    def __init__(self, *args, **kwargs):
        super(TestRepeatEosBytes, self).__init__(*args, **kwargs)
        self.struct_class = RepeatEosBytes
        self.src_filename = 'src/repeat_until_process.bin'

    def test_check_longer_io(self):
        r = RepeatEosBytes()

        r.records = [
            b"\xE8\xBA\xAA\xAA\xAA",
            b"\xFA\x9E\xB8\xAA\xAA",
            b"\xAA\x55\x55\x55\x55",
        ]

        r._check()

        ks_io = KaitaiStream(io.BytesIO(bytearray(15 + 3)))
        with self.assertRaisesRegex(ConsistencyError, u"^Check failed: records, expected: 0, actual: 3$"):
            r._write(ks_io)
